MODULE_NOT_FOUND for the entry point file (nodejs 20.9)

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
After upgrading Node from **16.20.2** to **20.9.0** a compiled binary fails to run.
The binary requires access to source code.

```
arm-adapters-1  | Error: Cannot find module '/app/dist/index.js'
arm-adapters-1  |     at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
arm-adapters-1  |     at Module._load (node:internal/modules/cjs/loader:901:27)
arm-adapters-1  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
arm-adapters-1  |     at node:internal/main/run_main_module:23:47 {
arm-adapters-1  |   code: 'MODULE_NOT_FOUND',
arm-adapters-1  |   requireStack: []
arm-adapters-1  | }
arm-adapters-1  | 
arm-adapters-1  | Node.js v20.9.0
```

**/app/dist/index.js** is the entry point.

**What you expected to happen**:
I expect it to work.

**How to reproduce it (as minimally and precisely as possible)**:
Upgrade to Node 20.9.0

**Anything else we need to know?**:
If you place a folder with source code next to a compiled binary, the binary works well. 
A command to compile a binary:
```
nexe --input dist/index.js --output app_binary --build --make="-j16" --python=$(which python3)
```


**Environment**:
- Platform(OS/Version): Ubuntu 22.04.3 
- Host Node Version: 20.9
- Target Node Version: 20.9
- Nexe version: 4.0.0-rc.2
- Python Version: 3.10

--- a/src/fs/patch.ts
+++ b/src/fs/patch.ts
@@ -82,7 +82,9 @@ function shimFs(binary: NexeHeader, fs: typeof import('fs') = require('fs')) {
     if (returningArray == null) returningArray = Array.isArray(original.call(this, ''))
     const res = internalModuleReadFile.call(this, original, ...args)
     return returningArray && !Array.isArray(res)
-      ? [res, /"(main|name|type|exports|imports)"/.test(res)]
+      ? res === ''
+        ? []
+        : [res, /"(main|name|type|exports|imports)"/.test(res)]
       : res
   }
   patches.internalModuleStat = function (this: any, original: any, ...args: any[]) {

--- a/src/patches/third-party-main.ts
+++ b/src/patches/third-party-main.ts
@@ -55,13 +55,22 @@ export default async function main(compiler: NexeCompiler, next: () => Promise<v
   })
 
   const fileLines = file.contents.toString().split('\n')
-  fileLines.splice(
-    location.start.line,
-    0,
-    '{{ file("lib/fs/bootstrap.js") }}' +
-      '\n' +
-      (semverGt(version, '11.99') ? 'expandArgv1 = false;\n' : '')
-  )
+  if (semverGt(version, '19.99')) {
+    await compiler.replaceInFileAsync(
+      'lib/internal/modules/cjs/loader.js',
+      "'use strict';",
+      "'use strict';\n" + '{{ file("lib/fs/bootstrap.js") }}' + '\n'
+    )
+    fileLines.splice(location.start.line, 0, 'expandArgv1 = false;')
+  } else {
+    fileLines.splice(
+      location.start.line,
+      0,
+      '{{ file("lib/fs/bootstrap.js") }}' +
+        '\n' +
+        (semverGt(version, '11.99') ? 'expandArgv1 = false;\n' : '')
+    )
+  }
   file.contents = fileLines.join('\n')
 
   if (semverGt(version, '11.99')) {

